include/aws/crt/iot/SecureTunnel.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/crt/io/Bootstrap.h>
+#include <aws/crt/io/SocketOptions.h>
+#include <aws/iotdevice/secure_tunneling.h>
+
+namespace Aws
+{
+    namespace Crt
+    {
+        namespace Iot
+        {
+            // Callbacks
+            using OnSecureTunnelingConnectionComplete = std::function<void(void)>;
+            using OnSecureTunnelingSendDataComplete = std::function<void(int errorCode)>;
+            using OnSecureTunnelingDataReceive = std::function<void(const ByteBuf &data)>;
+            using OnSecureTunnelingStreamStart = std::function<void(int32_t streamId)>;
+            using OnSecureTunnelingStreamReset = std::function<void(void)>;
```

Comment:
Depending on the API's intent on what to manage, either stream ID's should be on all of these callbacks, or none of them. If the underlying handling is tracking the active valid stream ID, and properly ignores all messages of a non-matching stream IDs*, then above it stream ID has no value to pass along since it will only ever see valid data, and reset messages.

*StreamStart which indicates there is a new valid StreamID, implicitly closing the previous one